lift/Readme.md

```diff
@@ -2,275 +2,378 @@
+</details>
+<details>
+<summary>UpdateEnum</summary>
 
-While in Prismas model the declarative datamodel is the source of truth for migrations, Active Record takes the imperative migrations as the source of truth for
-migrations. Therefore one can call the imperative migrations also **manual migrations**.
+**TODO**
 
-### Datamodel Snapshots
+</details>
+<details>
+<summary>DeleteEnum</summary>
 
-As mentioned in [Declarative migration](#how-does-naming-of-migrations-work), the user adjusts the `datamodel.prisma` file, which results in a new migration
-folder when running `prisma migrate`. In order for the migration engine to know which changes need to be applied, it needs a "copy" of the datamodel. As this
-"copy" describes the datamodel in a specific point in time, we also refer to it as a "snapshot".
+**TODO**
 
-### Migration Engine
+</details>
 
-The migration engine is the heart of the whole migration system. It manages the connection to the database and performs the actual changes in the database
-schema. It is written in Rust and will be remotely controlled from the Prisma CLI written in JavaScript.
+### Hook
 
-### Migration History
+> ⚠ This is not implemented yet.
 
-When a migration is running or has been executed successfully, Prisma writes this information into a table or collection in the database. This table is also
-known as the "Migration History". It includes all successful and failed migrations of the specific Prisma Project. The benefits of storing this migration
-history are the following:
+A hook is a custom shell script that runs either before or after the migration. They are defined by `before.sh` and `after.sh`. Hooks give you full control over
+your migrations. You can write hooks to:
 
-- Prisma can make sure, that the local migrations are in sync with the databases migrations
-- Prisma can ensure, that migrations don't get run twice
-- Prisma can give an overview in a GUI interface about the currently running migration and migrations that already have been running
-- Prisma can rollback migrations when there has been an error
+- Ensure the data follows new constraints like `unique` or `non null`.
+- Add specific database primitives like functions in Postgres, which are not yet supported by Prisma.
+- Seed the database with data after a migration has been executed.
 
-## Detailed design
+## Architecture
 
-As mentioned in the [basic example](#basic-example), a simple `prisma migrate` is needed to migrate the database and create a new migration.
+Lift has 2 parts: a **Lift Client** and a **Lift Engine**. By default, the Lift Engine runs locally as a
+[sidecar process](https://blog.davemdavis.net/2018/03/13/the-sidecar-pattern/). This makes it easy to get started. As your team's data requirements become more
+complex, you may prefer to handle your migrations on a remote machine where you have fine-grained access control over migrations.
```

Comment:
"remote machine" / remote migration engine is not implemented yet and was removed elsewhere in the spec.